### PR TITLE
snap: refresh snap for `base: core22`

### PR DESCRIPTION
**Pull request by Saviq**
This refreshes the snap packaging so that https://snapcraft.io/sccache can be served better.

### CODE REVIEW

**Comment by Saviq** [comment]
This is now building here:

https://launchpad.net/~saviq/+snap/edge - it failed on s390x and ppc64el due to dependency issues.

I will follow up with a rework of the configuration hook to work with a configuration file rather than the environment.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+base: core22
+adopt-info: sccache
 
 summary: sccache is ccache with cloud storage
 description: |
-    sccache is a ccache-like tool. It is used as a compiler wrapper and avoids
-    compilation when possible, storing a cache in a remote storage using the
-    Amazon Simple Cloud Storage Service (S3) API, the Google Cloud Storage (GCS)
-    API, or Redis.
+    sccache is a [ccache](https://ccache.dev/)-like compiler caching tool. It
```
Comment (sylvestre) [inline snap/snapcraft.yaml:7]:
```suggestion
    sccache is a ccache-like compiler caching tool. It
```

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+    is used as a compiler wrapper and avoids compilation when possible, storing
+    cached results either on [local disk](#local) or in one of [several cloud
+    storage backends](#storage-options).
+
+    sccache includes support for caching the compilation of C/C++ code,
+    [Rust](docs/Rust.md), as well as NVIDIA's CUDA using
```
Comment (sylvestre) [inline snap/snapcraft.yaml:13]: This link will work ?
(docs/Rust.ml ?)

**Comment by sylvestre** [comment]
About the s390x port, see https://github.com/mozilla/sccache/issues/1442

File: snap/snapcraft.yaml (hunk 2 above)
Comment (Saviq) [inline snap/snapcraft.yaml:13]:
```suggestion
    [Rust](https://github.com/mozilla/sccache/blob/main/docs/Rust.md), as well as NVIDIA's CUDA using
```

**Comment by sylvestre** [comment]
yeah, I will ask the team to enable it, thanks :)

**Comment by sylvestre** [comment]
Uploading them on tagged releases works for me

**Comment by Saviq** [comment]
> Could you please make sure it doesn't run as root in snap? thanks

I've just dropped the daemon altogether. From what I understand about how sccache operates, it doesn't make sense to have a central server.

File: .github/workflows/snap.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    timeout-minutes: 45
+
+    strategy:
+      matrix:
+        platform:
+        - amd64
+        - armhf
+        - arm64
+        # Broken dependencies: - ppc64el
+        # Broken environment: - s390x
```
Comment (sylvestre) [inline .github/workflows/snap.yml:20]:
```suggestion
        # https://github.com/mozilla/sccache/issues/1442
        # Broken environment: - s390x
```

**Comment by Saviq** [comment]
@sylvestre I could move the build-snap to @release to avoid that, but then it could only build for amd64.

Your call!

**Comment by sylvestre** [comment]
@Saviq "I could move the build-snap to @Release to avoid that, but then it could only build for amd64."
=> yeah, this is fine, thanks

**Comment by Saviq** [comment]
> yeah, this is fine, thanks

Done.

**Comment by sylvestre** [comment]
I think we fixed the server side configuration but we have an error:
```

Creating new base instance 'base-instance-snapcraft-buildd-base-v00--a12c0d0071c13c372948' from instance.
readlink: missing operand
Try 'readlink --help' for more information.
Run REVIEW_OPTS=()
Errors
------
 - lint-snap-v2:confinement_classic
	(NEEDS REVIEW) confinement 'classic' not allowed. If your snap needs classic confinement to function, please make a request for this snap to use classic by creating a new topic in the forum using the 'store-requests' category and detail the technical reasons why classic is required.
	https://forum.snapcraft.io/t/process-for-reviewing-classic-confinement-snaps/1460
sccache_v0.4.1-25-g7cba138_amd64.snap: FAIL
```

**Comment by Saviq** [comment]
> I think we fixed the server side configuration but we have an error:

Fixed, sorry!

**Comment by sylvestre** [comment]
no worries :)

**Comment by sylvestre** [comment]
@Saviq sorry for the dumb question but how do we publish it on snap ?
I see see the old release
https://snapcraft.io/sccache

**Comment by Saviq** [comment]
> @Saviq sorry for the dumb question but how do we publish it on snap ? I see see the old release https://snapcraft.io/sccache

Hey, so https://snapcraft.io/docs/releasing-your-app manually, but if you set the `SNAPCRAFT_TOKEN` secret, we can upload from CI.

**Comment by Saviq** [comment]
@sylvestre to get builds from `main`, best probably hook it up under https://snapcraft.io/sccache/builds - you'll then just need to move the `edge` revisions to `beta`/`candidate`/`stable` as appropriate, here: https://snapcraft.io/sccache/releases

**Pull request merged**